web/tests/unread.test.js

```diff
@@ -527,32 +533,45 @@ test("mentions", () => {
         unread: true,
     };
 
+    const private_mention_me_message = {
+        id: 19,
+        type: "private",
+        display_recipient: [{id: anybody.user_id}],
+        mentioned: true,
+        mentioned_me_directly: true,
+        unread: true,
+    };
+
     unread.process_loaded_messages([
         already_read_message,
         mention_me_message,
         mention_all_message,
         muted_mention_all_message,
         muted_direct_mention_message,
+        private_mention_me_message,
     ]);
 
     counts = unread.get_counts();
-    assert.equal(counts.mentioned_message_count, 3);
+    assert.equal(counts.mentioned_message_count, 4);
     assert.deepEqual(unread.get_msg_ids_for_mentions(), [
         mention_me_message.id,
         mention_all_message.id,
         muted_direct_mention_message.id,
+        private_mention_me_message.id,
     ]);
     assert.deepEqual(unread.get_all_msg_ids(), [
         mention_me_message.id,
         mention_all_message.id,
         muted_mention_all_message.id,
         muted_direct_mention_message.id,
+        private_mention_me_message.id,
     ]);
-    test_notifiable_count(counts.home_unread_messages, 3);
+    test_notifiable_count(counts.home_unread_messages, 5);
```

Comment:
@timabbott, I am not sure why adding a private message is increasing the unread counter by 2(it is incrementing correctly the mentioned message count by 1). I tried debugging it but couldn't find the reason, am I missing something or is it a bug?

Reply:
I think this is a bug; nice find.
`calculate_notifiable_count` does this, which clearly doesn't consider the possibility of a mention in a private message:
`        new_message_count = res.mentioned_message_count + res.private_message_count;                    `